File: MyWorkItemFrontend/src/types/workItem.ts
export enum WorkItemStatus {
  Todo = 'Todo',
  InProgress = 'InProgress',
  Done = 'Done',
}

export enum Priority {
  Low = 'Low',
  Medium = 'Medium',
  High = 'High',
}

export interface WorkItem {
  id: number;
  title: string;
  description?: string;
  status: WorkItemStatus;
  priority: Priority;
  createdAt: string;
  updatedAt?: string;
}

export interface CreateWorkItemDto {
  title: string;
  description?: string;
  status: WorkItemStatus;
  priority: Priority;
}

export interface UpdateWorkItemDto {
  title: string;
  description?: string;
  status: WorkItemStatus;
  priority: Priority;
}

File: MyWorkItemAPI/MyWorkItemAPI/Models/WorkItem.cs
namespace MyWorkItemAPI.Models;

public enum WorkItemStatus
{
    Todo,
    InProgress,
    Done
}

public enum Priority
{
    Low,
    Medium,
    High
}

public class WorkItem
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public WorkItemStatus Status { get; set; } = WorkItemStatus.Todo;
    public Priority Priority { get; set; } = Priority.Medium;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }
}

public class CreateWorkItemDto
{
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public WorkItemStatus Status { get; set; } = WorkItemStatus.Todo;
    public Priority Priority { get; set; } = Priority.Medium;
}

public class UpdateWorkItemDto
{
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public WorkItemStatus Status { get; set; }
    public Priority Priority { get; set; }
}
